### PR TITLE
add `-r` option for recursively signing multi-arch images

### PR DESCRIPTION
**Pull request by dekkagaijin**
Signed-off-by: Jake Sanders <jsand@google.com>

### CODE REVIEW

File: cmd/cosign/cli/sign.go [hunk 1]
```diff
@@ -153,21 +157,25 @@ func SignCmd(ctx context.Context, so SignOpts,
...
+	if recursive && get.MediaType.IsIndex() {
+		idx, err := get.ImageIndex()
+		if err != nil {
+			return err
+		}
+		idxManifest, err := idx.IndexManifest()
+		if err != nil {
+			return err
+		}
+		for _, manifest := range idxManifest.Manifests {
```
Comment (dlorenc) [inline cmd/cosign/cli/sign.go:175]: I always forget - is it possible to create an image that recurses even further? An index of indexes?

Comment (dekkagaijin) [reply cmd/cosign/cli/sign.go:175]: not iiuc @jonjohnsonjr

Comment (jonjohnsonjr) [reply cmd/cosign/cli/sign.go:175]: it is

Comment (dekkagaijin) [reply cmd/cosign/cli/sign.go:175]: Talked with him and apparently recursive indexes _can_ exist 🤮

Comment (dekkagaijin) [reply cmd/cosign/cli/sign.go]: walking the tree now

Comment (jonjohnsonjr) [reply cmd/cosign/cli/sign.go]: it hurts me that the flag is recursive but the implementation is not

Comment (dekkagaijin) [reply cmd/cosign/cli/sign.go]: your anguish sustains me ❤️

**Review by dlorenc — APPROVED** [review]

**Pull request merged**